kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -294,72 +294,73 @@ def solc_compile(contract_file: Path, profile: bool = False) -> Dict[str, Any]:
+    claim, _ = build_claim(claim_name, init_cterm, final_cterm)
+    return claim
+
+
+def _calldata_cell(contract: Contract, method: Contract.Method) -> KInner:
+    contract_function_application_label = contract.klabel_method
+    klabel = method.production.klabel
+    assert klabel is not None
+    args = [
+        abstract_term_safely(KVariable('_###SOLIDITY_ARG_VAR###_'), base_name=f'V{name}') for name in method.arg_names
+    ]
+    return KApply(contract_function_application_label, [KApply(contract.klabel), KApply(klabel, args)])
+
+
+def _callvalue_cell(method: Contract.Method) -> KInner:
+    return (
+        intToken(0)
+        if not method.payable
+        else abstract_term_safely(KVariable('_###CALLVALUE###_'), base_name='CALLVALUE')
+    )
+
+
+def _default_claim(empty_config: KInner, contract_name: str) -> KClaim:
     init_term = _init_term(empty_config, contract_name)
-    init_terms = []
-    if calldata_cells:
-        for test_name, call_data, call_value in calldata_cells:
-            failing = test_name.startswith('testFail')
-            refined_subst = {'CALLDATA_CELL': call_data, 'CALLVALUE_CELL': call_value}
-            claim_name = test_name.replace('_', '-')
-            init_terms.append((claim_name, substitute(init_term, refined_subst), failing))
-    else:
-        init_terms.append((contract_name.lower(), init_term, False))
-    final_term = _final_term(empty_config, contract_name)
-    final_cterm = CTerm(final_term)
+    init_cterm = _init_cterm(init_term)
+    final_cterm = _final_cterm(empty_config, contract_name, failing=False)
+    claim, _ = build_claim(contract_name.lower(), init_cterm, final_cterm)
+    return claim
+
+
+def _init_cterm(init_term: KInner) -> CTerm:
     key_dst = KEVM.loc(KToken('FoundryCheat . Failed', 'ContractAccess'))
     dst_failed_prev = KEVM.lookup(KVariable('CHEATCODE_STORAGE'), key_dst)
-    dst_failed_post = KEVM.lookup(KVariable('CHEATCODE_STORAGE_FINAL'), key_dst)
-    claims: List[KClaim] = []
-    foundry_success = Foundry.success(KVariable('STATUSCODE_FINAL'), dst_failed_post)
-    for claim_id, i_term, failing in init_terms:
-        i_cterm = CTerm(i_term).add_constraint(mlEqualsTrue(KApply('_==Int_', [dst_failed_prev, KToken('0', 'Int')])))
-        if not failing:
-            f_cterm = final_cterm.add_constraint(mlEqualsTrue(foundry_success))
-        else:
-            f_cterm = final_cterm.add_constraint(mlEqualsTrue(notBool(foundry_success)))
-        claim, _ = build_claim(claim_id, i_cterm, f_cterm)
-        claims.append(claim)
-    return claims
+    return CTerm(init_term).add_constraint(mlEqualsTrue(KApply('_==Int_', [dst_failed_prev, KToken('0', 'Int')])))
+
+
+def _init_term_with_calldata(
+    init_term: KInner,
+    calldata: KInner,
+    callvalue: KInner,
+) -> KInner:
+    subst = {'CALLDATA_CELL': calldata, 'CALLVALUE_CELL': callvalue}
+    return substitute(init_term, subst)
```

Comment:
Seems like this doesn't need its own function.

Reply:
Done.